osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs

```diff
@@ -421,13 +422,42 @@ private static void applyAnchor(Drawable drawable, Anchor anchor)
             drawable.Position -= drawable.AnchorPosition - previousAnchor;
         }
 
-        private static void applyOrigin(Drawable drawable, Anchor origin)
+        private static void applyOrigin(Drawable drawable, Anchor screenSpaceOrigin)
```

Comment:
I don't think we've ever used the concept of a "screen-space" origin previously. Can you explain what this means?

Reply:
The point of the "screen space origin" thing is counteracting flips.

Consider the following: if a drawable with centre-left origin is flipped horizontally, its "effective" "screen space" origin is *centre-right*.

The idea is that if you move a flipped drawable to the centre-right area of the screen, its centre-right anchor should stick to the centre-right of its parenting container *despite* the fact that it's flipped. So you want it to have a *centre-left* origin locally to it, but also for it to have negative X scale. 

Does that make more sense?